File: packages/csharp/HtmlToMarkdown/Metadata/ImageType.cs
using System.Text.Json.Serialization;
using HtmlToMarkdown.Serialization;

namespace HtmlToMarkdown.Metadata;

/// <summary>
/// Image source classification for proper handling and processing.
/// Determines whether an image is embedded (data URI), inline SVG, external, or relative.
/// </summary>
[JsonConverter(typeof(JsonPropertyNameEnumConverter<ImageType>))]
public enum ImageType
{
    /// <summary>
    /// Data URI embedded image (base64 or other encoding).
    /// </summary>
    [JsonPropertyName("data_uri")]
    DataUri,

    /// <summary>
    /// Inline SVG element.
    /// </summary>
    [JsonPropertyName("inline_svg")]
    InlineSvg,

    /// <summary>
    /// External image URL (http/https).
    /// </summary>
    [JsonPropertyName("external")]
    External,

    /// <summary>
    /// Relative image path.
    /// </summary>
    [JsonPropertyName("relative")]
    Relative,
}

File: packages/java/src/main/java/dev/kreuzberg/htmltomarkdown/metadata/ImageType.java
package dev.kreuzberg.htmltomarkdown.metadata;

/**
 * Image source classification for proper handling and processing.
 *
 * <p>Determines whether an image is embedded (data URI), inline SVG, external, or relative.
 *
 * @since 2.13.0
 */
public enum ImageType {
  /** Data URI embedded image (base64 or other encoding). */
  DATA_URI("data_uri"),

  /** Inline SVG element. */
  INLINE_SVG("inline_svg"),

  /** External image URL (http/https). */
  EXTERNAL("external"),

  /** Relative image path. */
  RELATIVE("relative");

  private final String value;

  ImageType(String value) {
    this.value = value;
  }

  /**
   * Get the string representation of this image type.
   *
   * @return the type string ("data_uri", "inline_svg", "external", or "relative")
   */
  public String getValue() {
    return value;
  }

  /**
   * Parse an image type from string value.
   *
   * @param s the type string
   * @return the {@code ImageType}
   * @throws IllegalArgumentException if the string is null or unknown
   */
  public static ImageType parse(String s) {
    if (s == null) {
      throw new IllegalArgumentException("ImageType string cannot be null");
    }
    return switch (s.toLowerCase()) {
      case "data_uri" -> DATA_URI;
      case "inline_svg" -> INLINE_SVG;
      case "external" -> EXTERNAL;
      case "relative" -> RELATIVE;
      default -> throw new IllegalArgumentException("Unknown ImageType: " + s);
    };
  }

  @Override
  public String toString() {
    return value;
  }
}
